docs: update component ownership guidelines and templates (#2641)
open-telemetry/opentelemetry-js-contrib: .github/ISSUE_TEMPLATE/instrumentation_request.md

-->

### Which library do you want to instrument?
<!--
A clear and concise description of why this library should be instrumented. Ex. I'm always frustrated when [...]
-->

### Code Owners

<!--
New Instrumentations MUST have at least 2 code-owners that will maintain the instrumentation long-term.
Any code-owners listed MUST fulfill all criteria laid out in the checklist below.
-->

- Owner 1
  - [ ] I am a [member of the OpenTelemetry Organization](https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#member)
  - [ ] I have read [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) and understand the responsibilities of a component owner
  - [ ] I agree to follow and uphold the [mission, vision and values](https://github.com/open-telemetry/community/blob/main/mission-vision-values.md) of the OpenTelemetry project
  - [ ] I understand that the component may be subject to the [OpenTelemetry Semantic Conventions](https://github.com/open-telemetry/semantic-conventions) and agree to follow the rules set out therein
- Owner 2
  <!-- instruct any co-owners to complete the checklist above and comment it on this issue -->

>   - [ ] Commented the checklist below and has fulfilled the criteria for becoming a component owner

Reviewer: "Commented the checklist below" sounds a little weird, I was confused by what it was trying to say.
Do you mean something like "Commented on the issue confirming it has fulfilled all criteria for becoming a component owner" ?

Author: Haha, yes I was confused when writing it too and then gave up at some point 😬 
Thank you for the suggestion - I think your version is better 🙂 

```suggestion
  - [ ] Commented on the issue confirming they have fulfilled all criteria for becoming a component owner
```

>   - [ ] Commented the checklist below and has fulfilled the criteria for becoming a component owner

Author: Haha, yes I was confused when writing it too and then gave up at some point 😬 
Thank you for the suggestion - I think your version is better 🙂 

```suggestion
  - [ ] Commented on the issue confirming they have fulfilled all criteria for becoming a component owner
```
